### DISCUSSION TITLE
Allow sortable column titles to render components

### DISCUSSION THREAD

**Discussion by tcbarrett** [TanStack/table#5711] [Ideas]
From React Sorting example: https://tanstack.com/table/v8/docs/framework/react/examples/sorting

```
                        title={
                          header.column.getCanSort()
                            ? header.column.getNextSortingOrder() === 'asc'
                              ? 'Sort ascending'
                              : header.column.getNextSortingOrder() === 'desc'
                                ? 'Sort descending'
                                : 'Clear sort'
                            : undefined
                        }
```

I would useful if those text strings could be components to allow re-rendering of the titles - for such things as translations and other dynamic custom sorting requirements.

**Comment by KevinVandy** [comment]
Why can't they be components? You're in charge of - this code in your app. Just use components then.

**Reply by tcbarrett** [reply]
Because passing in a component doesn't render.

[Codebox Example](https://codesandbox.io/p/devbox/jovial-moon-47wgjd?workspaceId=83cb73ec-6b96-4596-9c1d-e89e0f3235df&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cm01g05w100083b6k11ve4tne%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cm01g05w100033b6k2n2datvu%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cm01g05w100053b6kzjtm9q5r%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cm01g05w100073b6kxf6hh2ux%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cm01g05w100033b6k2n2datvu%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm01g05w100023b6kd7qqop4b%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cm01g05w100033b6k2n2datvu%2522%252C%2522activeTabId%2522%253A%2522cm01g05w100023b6kd7qqop4b%2522%257D%252C%2522cm01g05w100073b6kxf6hh2ux%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm01g05w100063b6k6564vjoh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%257D%255D%252C%2522id%2522%253A%2522cm01g05w100073b6kxf6hh2ux%2522%252C%2522activeTabId%2522%253A%2522cm01g05w100063b6k6564vjoh%2522%257D%252C%2522cm01g05w100053b6kzjtm9q5r%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm01g05w100043b6kqaiwckkd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522cm01g05w100053b6kzjtm9q5r%2522%252C%2522activeTabId%2522%253A%2522cm01g05w100043b6kqaiwckkd%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Image showing hover state issue:
<img width="231" alt="Screenshot 2024-08-19 at 21 27 12" src="https://github.com/user-attachments/assets/684aed60-87b5-47a7-9c57-15cbd9ea51a3">

Maybe I missed something obvious?
